src/onyx/messaging/aeron.clj

```diff
@@ -72,99 +103,99 @@
-             :completion-stream-id completion-stream-id
-             :accept-send-fut accept-send-fut
-             :accept-acker-fut accept-acker-fut
-             :accept-completion-fut accept-completion-fut
-             :send-subscriber send-subscriber
-             :acker-subscriber acker-subscriber
-             :completion-subscriber completion-subscriber
-             :release-ch release-ch 
-             :context ctx 
-             :aeron aeron)))
-
-  (stop [{:keys [aeron send-subscriber acker-subscriber 
-                 completion-subscriber accept-send-fut accept-acker-fut
-                 accept-completion-fut release-ch] :as component}]
-    (taoensso.timbre/info "Stopping Aeron")
+             :bind-addr bind-addr 
+             :external-addr external-addr
+             :resources (atom nil) 
+             :release-ch release-ch)))
 
+  (stop [{:keys [aeron resources release-ch] :as component}]
+    (taoensso.timbre/info "Stopping Aeron")
     (try 
-      (future-cancel accept-send-fut)
-      (future-cancel accept-acker-fut)
-      (future-cancel accept-completion-fut)
-      (when send-subscriber (.close send-subscriber))
-      (when acker-subscriber (.close acker-subscriber))
-      (when completion-subscriber (.close completion-subscriber))
-      (when aeron (.close aeron))
+      (when-let [rs @resources]
+        (let [{:keys [conn
+                      send-subscriber 
+                      acker-subscriber 
+                      completion-subscriber 
+                      accept-send-fut 
+                      accept-acker-fut
+                      accept-completion-fut]} rs] 
+          (future-cancel accept-send-fut)
+          (future-cancel accept-acker-fut)
+          (future-cancel accept-completion-fut)
+          (when send-subscriber (.close send-subscriber))
+          (when acker-subscriber (.close acker-subscriber))
+          (when completion-subscriber (.close completion-subscriber))
+          (when conn (.close conn)))
+        (reset! resources nil))
       (close! (:release-ch component))
-      ;; FIXME, need to startup and shutdown aeron resources properly
-      ;; however, this can't be done per connection
-      ;(CloseHelper/quietClose (:driver component))
+      (catch Exception e (fatal e)))
 
-      (catch Exception e
-        (fatal e)))
-
-    (assoc component :driver nil :channel nil :release-ch nil :aeron nil 
-           :accept-send-fut nil :accept-acker-fut nil :accept-completion-fut nil 
-           :send-subscriber nil :acker-subscriber nil :completion-subscriber nil)))
+    (assoc component :bind-addr nil :external-addr nil :site-resources nil :release-ch nil)))
 
 (defn aeron [opts]
   (map->AeronConnection {:opts opts}))
 
-(defmethod extensions/send-peer-site AeronConnection
+(defmethod extensions/peer-site AeronConnection
   [messenger]
-  {:channel (:channel messenger)
-   :stream-id (:send-stream-id messenger)})
+  {:aeron/bind-addr (:bind-addr messenger)
```

Comment:
Without both, the port assignment gets messy for containers and it must be reproducible by the other peers. 

Imagine two containers on the same physical host:
A internal 192.168.1.1 external 10.0.0.1
B internal 192.168.1.2 external 10.0.0.1

Without both external and internal IPs, port allocation will be A 40200, B, 40201. This will be annoying and non deterministic when forwarding ports from the physical host to the container.

After writing it, I now realise the flaw with this approach is that the advertised port will be different to the forwarded port.

Maybe we need some kind of per peer advertised min port / max port too. Will have to think about it some more. For now we can just use the external ip so ports map directly between the hosts and the containers.